contracts/contracts/FleekERC721.sol

```diff
@@ -367,9 +414,52 @@ contract FleekERC721 is Initializable, ERC721Upgradeable, FleekAccessControl {
         _requireMinted(tokenId);
         require(_accessPoints[apName].owner == address(0), "FleekERC721: AP already exists");
 
-        _accessPoints[apName] = AccessPoint(tokenId, 0, false, false, msg.sender);
-
         emit NewAccessPoint(apName, tokenId, msg.sender);
+
+        if (_apps[tokenId].accessPointAutoApprovalSettings) {
+            // Auto Approval is on.
+            _accessPoints[apName] = AccessPoint(tokenId, 0, false, false, msg.sender, AccessPointStatus.APPROVED);
+
+            emit ChangeAccessPointStatus(apName, tokenId, AccessPointStatus.APPROVED, msg.sender);
+        } else {
+            // Auto Approval is off. Should wait for approval.
+            _accessPoints[apName] = AccessPoint(tokenId, 0, false, false, msg.sender, AccessPointStatus.DRAFT);
+            emit ChangeAccessPointStatus(apName, tokenId, AccessPointStatus.DRAFT, msg.sender);
+        }
+    }
+
+    /**
+     * @dev Set approval settings for an access point.
+     * It will add the access point to the token's AP list, if `approved` is true.
+     *
+     * May emit a {ChangeAccessPointApprovalStatus} event.
+     *
+     * Requirements:
+     *
+     * - the tokenId must exist and be the same as the tokenId that is set for the AP.
+     * - the AP must exist.
+     * - must be called by a token controller.
+     */
+    function setApprovalForAccessPoint(
+        uint256 tokenId,
+        string memory apName,
+        bool approved
+    ) public requireTokenRole(tokenId, Roles.Controller) {
```

Comment:
Should we also require that current status DRAFT? That way it won't run if its already set or invalid.